src/components/services-view/services-view.controller.ts

```diff
@@ -2,155 +2,188 @@ import * as angular from 'angular';
-  }
-
-  public $onInit() {
-    this.ctrl.currentFilter = 'all';
-    this.ctrl.currentSubFilter = null;
-    this.ctrl.expandSubCatRow = ServicesViewController.NO_SUBCATEGORY_ROW;
-    this.ctrl.orderingPanelvisible = false;
-    this.ctrl.loading = _.isEmpty(this.ctrl.services);
-
-    this.updateAll();
-
-    this.$scope.$on('cancelOrder', () => {
-      this.ctrl.closeOrderingPanel();
-    });
- }
-
-  public $onChanges(onChangesObj: angular.IOnChangesObject) {
-    if (onChangesObj.categories && !onChangesObj.categories.isFirstChange()) {
-      this.ctrl.categories = onChangesObj.categories.currentValue;
-      this.updateCategories();
+    static $inject = ['Constants', 'Catalog', '$filter', '$scope'];
+
+    static readonly NO_SUBCATEGORY_ROW = -1;
+
+    public ctrl: any = this;
+    public cardViewConfig: any;
+    private constants: any;
+    private catalog: any;
+    private $filter: any;
+    private $scope: any;
+    private subCatMaxRowLength: number = 10;
+    private serviceClassesLoaded = false;
+    private imageStreamsLoaded = false;
+
+    constructor(constants: any, catalog: any, $filter: any, $scope: any) {
+        this.cardViewConfig = {
+            selectItems: false,
+            showSelectBox: false,
+            onClick: this.handleClick
+        };
+        this.constants = constants;
+        this.catalog = catalog;
+        this.$filter = $filter;
+        this.$scope = $scope;
+        this.ctrl.loading = true;
     }
-    if (onChangesObj.services && !onChangesObj.services.isFirstChange()) {
-      this.ctrl.services = onChangesObj.services.currentValue;
-      this.updateServices();
-      this.ctrl.loading = false;
+
+    public $onInit() {
+        this.ctrl.allItems = [];
+        this.ctrl.currentFilter = 'all';
+        this.ctrl.currentSubFilter = null;
+        this.ctrl.expandSubCatRow = ServicesViewController.NO_SUBCATEGORY_ROW;
+        this.ctrl.orderingPanelvisible = false;
+        this.ctrl.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+        this.ctrl.subCategories = this.getSubCategories('all');
+
+        this.updateAll();
+
+        this.$scope.$on('cancelOrder', () => {
+            this.ctrl.closeOrderingPanel();
+        });
     }
-  }
-
-  public filterByCategory(category: string, subCategory: string, updateSubCategories: boolean) {
-    if (category === 'all' && subCategory === 'all') {
-      this.ctrl.filteredServices = this.ctrl.origServices;
-    } else {
-      this.ctrl.filteredServices = this.$filter('filter')(this.ctrl.origServices, (service: any) => {
-        if (category !== 'all' && subCategory === 'all') {
-          return service.metadata.category === category;
-        } else if (category === 'all' && subCategory !== 'all') {
-          return service.metadata.subCategory === subCategory;
+
+    public $onChanges(onChangesObj: angular.IOnChangesObject) {
+        if (onChangesObj.serviceClasses && !onChangesObj.serviceClasses.isFirstChange()) {
```

Comment:
I believe the only way this.ctrl.loading will never be set to false (what we need to stop the spinners) if images streams or service classes are initially set to empty, and never updated again asynchronously.  After DataService.list sets those collections asynchronously (to empty obj. or otherwise) the flags (serviceClassesLoaded & imageStreamsLoaded) will be set to true and this.ctrl.loading will be set to false, thus stopping the spinners.